src/translator/translator.h

```diff
@@ -72,15 +66,19 @@ class Translate : public ModelTask {
     scorers_.resize(numDevices_);
     graphs_.resize(numDevices_);
 
-#if MMAP
     auto models = options->get<std::vector<std::string>>("models");
-    for(auto model : models) {
-      marian::filesystem::Path modelPath(model);
-      ABORT_IF(modelPath.extension() != marian::filesystem::Path(".bin"),
-              "Non-binarized models cannot be mmapped");
-      mmaps_.push_back(std::move(mio::mmap_source(model)));
+    if(options_->get<bool>("model-mmap")) {
```

Comment:
Consider using the getter with a default value as this might be safer/cleaner, i.e. `get<bool>("model-mmap", false)`.